components/bt/host/bluedroid/api/include/api/esp_gap_ble_api.h

```diff
@@ -17,2687 +17,3986 @@
+//@}
+//@{
+/** BLE_INCLUDED */
+    ESP_GAP_BLE_SC_OOB_REQ_EVT,                                  /*!< When Secure Connection Out-of-Band request, the event comes.  */
+    ESP_GAP_BLE_SC_CR_LOC_OOB_EVT,                               /*!< When Secure Connection create Out-of-Band data complete, the event comes. */
+    ESP_GAP_BLE_GET_DEV_NAME_COMPLETE_EVT,                       /*!< When getting bluetooth device name complete, the event comes. */
+//@}
+//@{
+/** if Bluetooth Low Energy Feature Periodic Advertising Synchronization Transfer Support */
+    ESP_GAP_BLE_PERIODIC_ADV_RECV_ENABLE_COMPLETE_EVT,           /*!< When set periodic advertising receive enable complete, the event comes. */
+    ESP_GAP_BLE_PERIODIC_ADV_SYNC_TRANS_COMPLETE_EVT,            /*!< When periodic advertising sync transfer complete, the event comes. */
+    ESP_GAP_BLE_PERIODIC_ADV_SET_INFO_TRANS_COMPLETE_EVT,        /*!< When periodic advertising set info transfer complete, the event comes. */
+    ESP_GAP_BLE_SET_PAST_PARAMS_COMPLETE_EVT,                    /*!< When set periodic advertising sync transfer params complete, the event comes. */
+    ESP_GAP_BLE_PERIODIC_ADV_SYNC_TRANS_RECV_EVT,                /*!< When periodic advertising sync transfer received, the event comes. */
+//@}
+//@{
+/** direct test mode */
+    ESP_GAP_BLE_DTM_TEST_UPDATE_EVT,                             /*!< When direct test mode state changes, the event comes. */
+//@}
+//@{
+/** BLE_INCLUDED */
+    ESP_GAP_BLE_ADV_CLEAR_COMPLETE_EVT,                          /*!< When clear advertising complete, the event comes. */
+    ESP_GAP_BLE_SET_RPA_TIMEOUT_COMPLETE_EVT,                    /*!< When set the Resolvable Private Address (RPA) timeout completes, the event comes. */
+    ESP_GAP_BLE_ADD_DEV_TO_RESOLVING_LIST_COMPLETE_EVT,          /*!< When add a device to the resolving list completes, the event comes. */
+    ESP_GAP_BLE_VENDOR_CMD_COMPLETE_EVT,                         /*!< When vendor Host Controller Interface command complete, the event comes. */
+    ESP_GAP_BLE_SET_PRIVACY_MODE_COMPLETE_EVT,                   /*!< When set privacy mode complete, the event comes. */
+    ESP_GAP_BLE_EVT_MAX,                                         /*!< When maximum advertising event complete, the event comes. */
+//@}
 } esp_gap_ble_cb_event_t;
 
-#define ESP_GAP_BLE_CHANNELS_LEN     5                           /*!< channel length*/
-typedef uint8_t esp_gap_ble_channels[ESP_GAP_BLE_CHANNELS_LEN];
-
-/// This is the old name, just for backwards compatibility
-#define ESP_GAP_BLE_ADD_WHITELIST_COMPLETE_EVT ESP_GAP_BLE_UPDATE_WHITELIST_COMPLETE_EVT
-
-/// Advertising data maximum length
-#define ESP_BLE_ADV_DATA_LEN_MAX               31
-/// Scan response data maximum length
-#define ESP_BLE_SCAN_RSP_DATA_LEN_MAX          31
+#define ESP_GAP_BLE_CHANNELS_LEN     5                          /*!< Channel length includes three broadcast channels and two reserved channels. */
```

Comment:
Channel Map length. 37 bits (bit 0~36) meaningful. Includes 37 data channel and 3 reserved channels.